console/console-init/ui/src/modules/address-space/dialogs/CreateMessagingProject/MessagingProjectReview.tsx

```diff
@@ -8,34 +8,29 @@ import { Loading } from "use-patternfly";
 import { useQuery } from "@apollo/react-hooks";
 import { ADDRESS_SPACE_COMMAND_REVIEW_DETAIL } from "graphql-module/queries";
 import { AddressSpaceReview } from "modules/address-space/components";
+import { IMessagingProject } from "./CreateMessagingProject";
 
-export interface IReviewProps {
-  name?: string;
-  type?: string;
-  plan?: string;
-  namespace: string;
-  authenticationService: string;
+interface IMessagingProjectReviewProps {
+  projectDetail: IMessagingProject;
 }
 
-export const Review: React.FunctionComponent<IReviewProps> = ({
-  name,
-  type,
-  plan,
-  namespace,
-  authenticationService
+export const MessagingProjectReview: React.FunctionComponent<IMessagingProjectReviewProps> = ({
+  projectDetail
 }) => {
+  const { name, namespace, type, plan, authService } =
```

Comment:
is there any reason for adding && condition for same object. if you are facing run time undefined variable issue, use below syntax to avoid run time error in object destructurin.
 const { name, namespace, type, plan, authService } =projectDetail || {};